crates/chain/src/spk_iter.rs

```diff
@@ -195,6 +202,22 @@ mod test {
 
         let mut external_spk = SpkIterator::new(&no_wildcard_descriptor);
 
+        assert_eq!(external_spk.nth(0).unwrap(), (0, external_spk_0.clone()));
```

Comment:
Nit: this will give better errors. Will be really cool if we can tweak this for all asserts in the same PR.

```suggestion
        assert_eq!(external_spk.nth(0), Some((0, external_spk_0.clone())));
```